changelog.d/pa-2645.changed

```diff
@@ -0,0 +1,4 @@
+taint-mode: Added option `taint_assume_safe_comparisons`, disabled by default, that
+prevents comparison operators to propagate taint, so e.g. `tainted != "something"`
+will not be considered tainted. Note that this a syntactic check, if the operator
+is overloaded to perform a different operation this will not be detected.
```

Comment:
I don't know where you are in the doc writing, but I think it would be great to explain in what scenario this option is useful. Is it because comparing a variable against any other value is assumed to always be part of some sort of sanitization? When is it a safe assumption and when is it not?

Reply:
I'll clarify this better in the changelog, but this option simply makes the comparison expression `tainted != "something"` (or `tainted >= something`, etc) not tainted, however `tainted` will continue to be tainted. Previously, the result of comparing `tainted` against some other value, that is the resulting Boolean `true`/`false` was considered tainted.